src/pynwb/image.py

```diff
@@ -65,17 +71,25 @@ def __init__(self, **kwargs):
         if args_to_set['external_file'] is None and data is None:
             raise ValueError("Must supply either external_file or data to %s '%s'."
                              % (self.__class__.__name__, name))
-
         # data and unit are required in TimeSeries, but allowed to be None here, so handle this specially
         if data is None:
             kwargs['data'] = ImageSeries.DEFAULT_DATA
         if unit is None:
             kwargs['unit'] = ImageSeries.DEFAULT_UNIT
 
+        external_file_shape = get_data_shape(args_to_set['external_file'])
+        starting_frame_shape = get_data_shape(args_to_set['starting_frame'])
+        if external_file_shape is not None and starting_frame_shape is None:
+            if len(external_file_shape) == 1:
+                args_to_set['starting_frame'] = [0]
+        if external_file_shape != get_data_shape(args_to_set['starting_frame']):
+            warnings.warn(
```

Comment:
I would suggest to change this to an error, rather than just a warning

```suggestion
            raise ValueError(
```